sdk/loadtestservice/load-testing-rest/src/beginCreateOrUpdateTestRun.ts

```diff
@@ -0,0 +1,189 @@
+import { v4 as uuidv4 } from "uuid";
+import { TestRunStatusPoller, PolledOperationOptions } from "./models";
+import { AzureLoadTestingClient } from "./index.js";
+import { TestRunGet200Response } from "./responses";
+import { isUnexpected } from "./isUnexpected";
+
+/**
+ * Creates a poller to poll for test run status.
+ * @param client - The Load Testing client.
+ * @param options - The operation options.
+ * @returns A poller which can be called to poll until completion of the job.
+ */
+export async function beginTestRun(
+  client: AzureLoadTestingClient,
+  testId: string,
+  displayName: string,
+  polledOperationOptions: PolledOperationOptions = {}
+): Promise<TestRunStatusPoller> {
+  const testRunId = uuidv4(); // ID to be assigned to a testRun
+ // Creating the test run
+  const testRunCreationResult = await client.path("/test-runs/{testRunId}", testRunId).patch({
+    contentType: "application/merge-patch+json",
+    body: {
+      testId: testId,
+      displayName: displayName,
+      virtualUsers: 10,
+    },
+  });
+
+  if (isUnexpected(testRunCreationResult)) {
+    throw testRunCreationResult.body.error;
+  }
+
+  if (testRunCreationResult.body.testRunId === undefined)
+    throw new Error("Test Run ID returned as undefined.");
+
+  type Handler = (state: OperationState<TestRunGet200Response>) => void;
+
+  const state: OperationState<TestRunGet200Response> = {
+    status: "notStarted",
+  };
+
+  const progressCallbacks = new Map<symbol, Handler>();
+  const processProgressCallbacks = async (): Promise<void> =>
+    progressCallbacks.forEach((h) => h(state));
+  let resultPromise: Promise<TestRunGet200Response> | undefined;
+  let cancelJob: (() => void) | undefined;
+  const abortController = new AbortController();
+  const currentPollIntervalInMs = polledOperationOptions.updateIntervalInMs ?? 2000;
+
+  const poller: SimplePollerLike<OperationState<TestRunGet200Response>, TestRunGet200Response> = {
+    async poll(_options?: { abortSignal?: AbortSignalLike }): Promise<void> {
+      let getTestRunResult = await client.path("/test-runs/{testRunId}", testRunId).get();
+      if (isUnexpected(getTestRunResult)) {
+        throw getTestRunResult.body.error;
+      }
+
+      if (getTestRunResult.body.status === "FAILED") {
+        state.status = "failed";
+        state.error = new Error(getTestRunResult.body.status);
+      }
+
+      if (
+        getTestRunResult.body.status === "CANCELLING" ||
+        getTestRunResult.body.status === "CANCELLED"
+      ) {
+        state.status === "canceled";
+      }
+
+      if (getTestRunResult.body.status === "DONE") {
+        state.status = "succeeded";
+        state.result = getTestRunResult;
+      }
+
+      await processProgressCallbacks();
+
+      if (state.status === "canceled") {
+        throw new Error("Operation was canceled");
+      }
+      if (state.status === "failed") {
+        throw state.error;
+      }
```

Comment:
We don't want to throw errors in rest-level clients. The customer has to check `poller.getOperationState().status` to know the status of the operation.

Reply:
Updated.

Reply:
I don't see it being updated yet, I still see throw statements.

Reply:
Updated.